src/ResultRenderer/OutputFailureRenderer.php

```diff
@@ -32,11 +32,13 @@ public function __construct(StdOutFailure $result)
     public function render(ResultsRenderer $renderer)
     {
         return sprintf(
-            "  %s\n%s\n\n  %s\n%s\n",
+            "  %s\n%s\n\n  %s\n%s\n" . ($this->result->getWarnings() ? "\n  %s\n%s\n" : ""),
```

Comment:
string concatenation in the `sprintf` is a bit strange. Can you build it up separately? Maybe just one extra `%s` which is replaces with the warnings header and warnings or empty string if no warnings?